clientGQL_test.go

```diff
@@ -66,33 +76,159 @@ func ATestClient(t *testing.T, endpoint string) *ol.Client {
+	}
+	return data.String(), nil
+}
+
+func (t *FancyTemplater) JsonIndent(contents string) (string, error) {
+	output := bytes.NewBuffer([]byte{})
+	output.WriteString(contents)
+	contentsAsBytes := output.Bytes()
+	output.Reset()
+
+	err := json.Indent(output, contentsAsBytes, "", "  ")
+	if err != nil {
+		return "", fmt.Errorf("error during json indenting: %s", err)
+	}
+	return output.String(), nil
+}
+
+func NewFancyTemplater(templateDirs ...string) *FancyTemplater {
+	var templateFiles []string
+	for _, dir := range []string{"./testdata/templates"} {
+		err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() {
+				templateFiles = append(templateFiles, path)
+			}
+			return nil
+		})
+		if err != nil {
+			panic(fmt.Errorf("error during loading template files: %s", err))
+		}
+	}
+
+	output := FancyTemplater{}
+	tmpl, err := template.New("").Funcs(sprig.TxtFuncMap()).ParseFiles(templateFiles...)
+	if err != nil {
+		panic(fmt.Errorf("error during template initialization: %s", err))
+	}
+	output.rootTemplate = tmpl
+	return &output
+}
+
 type TestRequest struct {
+	templater *FancyTemplater
 	Request   string
 	Variables string
 	Response  string
 }
 
+func (t *TestRequest) RequestWithVariables() string {
+	jsonRequestWithVariables := fmt.Sprintf(`{%s, %s}`, t.Request, t.Variables)
+	if t.IsValidJson(jsonRequestWithVariables) {
+		return jsonRequestWithVariables
+	}
+	return ""
+}
+
+func (t *TestRequest) IsValidJson(data string) bool {
+	return json.Valid([]byte(data))
+}
+
+func (t *TestRequest) ParseRequest(rawRequest string) {
+	parsedRequest, _ := t.templater.ParseTemplatedString(rawRequest)
+	// queryValue, _ := strings.CutPrefix(rawRequest, queryPrefix)
+	jsonFormattedRequest := fmt.Sprintf("{%s}", parsedRequest)
+	if !t.IsValidJson(jsonFormattedRequest) {
+		panic(fmt.Errorf("test request could not be JSON formatted: %s", parsedRequest))
+	}
+	jsonFormattedRequest, err := t.templater.JsonIndent(fmt.Sprintf("{%s}", parsedRequest))
+	if err != nil {
+		panic(err)
+	}
+	t.Request = strings.TrimSpace(strings.Trim(jsonFormattedRequest, "{}"))
+}
+
+func (t *TestRequest) ParseVariables(rawVariables string) {
+	parsedVariables, err := t.templater.ParseTemplatedString(rawVariables)
+	if err != nil {
+		panic(err)
+	}
+
+	jsonFormattedVariableObject := strings.TrimSpace(parsedVariables)
+	jsonFormattedVariableObject, _ = strings.CutPrefix(jsonFormattedVariableObject, varPrefix)
+	if !t.IsValidJson(jsonFormattedVariableObject) {
+		panic(fmt.Errorf("test request could not be JSON formatted: %s", parsedVariables))
+	}
+	t.Variables = fmt.Sprintf("%s %s", varPrefix, jsonFormattedVariableObject)
+}
+
+func (t *TestRequest) ParseResponse(rawResponse string) {
+	parsedResponse, err := t.templater.ParseTemplatedString(rawResponse)
+	if err != nil {
+		panic(err)
+	}
+	if !t.IsValidJson(parsedResponse) {
+		panic(fmt.Errorf("invalid json: %s", parsedResponse))
```

Comment:
```suggestion
		panic(fmt.Errorf("test response could not be JSON formatted: %s", parsedResponse))
```

Just for consistency